baseapp/baseapp.go

```diff
@@ -948,6 +948,7 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, msgsV2 []protov2.Me
 		if err != nil {
 			return nil, errorsmod.Wrapf(err, "failed to execute message; message index: %d", i)
 		}
+		msgResponses = append(msgResponses, msgResult.MsgResponses...)
```

Comment:
Why remove the length check? Even though there should only be 1 entry in the response, isn't best to safe guard against anything more than 1 or zero?

Reply:
Hi, I reviewed the history, the [msgResult.MsgResponses](https://github.com/cosmos/cosmos-sdk/blob/59f67147f374afd13a4f1e65ab66402755c7b9d7/baseapp/baseapp.go#L828) length check was added in v0.46.x version, it was for compatibility with [legacyMsg](https://github.com/cosmos/cosmos-sdk/blob/59f67147f374afd13a4f1e65ab66402755c7b9d7/baseapp/baseapp.go#L790), if now we completely deprecate `legacyMsg`, I think it should be can delete it.

Reply:
While true, this ins't an expensive operation and some defensive programming isn't bad. I'm with bez to revert this and instead simply update the comment.